docs/router/mcp.mdx

```diff
@@ -0,0 +1,281 @@
+  </Card>
+  <Card title="get_schema" icon="code">
+    Provides the full GraphQL schema as a string. This helps AI models understand the entire API structure. This tool is only available if `expose_schema` is enabled.
+  </Card>
+</CardGroup>
+
+### Execution Tools
+
+<CardGroup cols={2}>
+  <Card title="execute" icon="play">
+    Executes arbitrary GraphQL queries or mutations against your API. This tool is only available if `enable_arbitrary_operations` is enabled, allowing AI models to craft and execute custom operations beyond predefined ones.
+  </Card>
+  <Card title="execute_*" icon="rocket">
+    For each GraphQL operation in your operations directory, the MCP server automatically generates a corresponding execution tool with the pattern `execute_<operation_name>` (e.g., `execute_get_users`).
+  </Card>
+</CardGroup>
+
+The operation execution tools provide a structured and controlled way for AI models to interact with your API:
+
+- **Tool naming**: Tools follow the pattern `execute_<operation_name>` (with operation names converted to snake_case)
+- **Tool schema**: Generated from your GraphQL operation's variables, ensuring type safety
+- **Tool description**: Inherited from your GraphQL operation descriptions, including operation name and additional context
+
+By default, all operations in your specified directory will be exposed as tools. Use the `exclude_mutations: true` configuration option to prevent mutation operations from being exposed if you want to ensure AI models can only read data.
+
+## Configuration
+
+To enable MCP in your Cosmo Router, add the following configuration to your `config.yaml`:
+
+```yaml
+mcp:
+  enabled: true
+  listen_addr: ":5025"
+  operations_dir: "operations"
+  graph_name: "my-graph"
+  exclude_mutations: true
+  enable_arbitrary_operations: false
+  expose_schema: true
+```
+
+### Configuration Options
+
+| Option | Description | Default |
+|--------|-------------|---------|
+| `enabled` | Enable or disable the MCP server | `false` |
+| `listen_addr` | The address where the MCP server will listen for requests | `:5025` |
+| `operations_dir` | Directory where GraphQL operations are stored | `operations` |
+| `graph_name` | The name of the graph to be used by the MCP server | `cosmo` |
+| `exclude_mutations` | Whether to exclude mutation operations from being exposed | `false` |
+| `enable_arbitrary_operations` | Whether to allow arbitrary GraphQL operations to be executed. <Warning>Security risk: Should only be enabled in secure, internal environments.</Warning> | `false` |
+| `expose_schema` | Whether to expose the full GraphQL schema. <Warning>Security risk: Should only be enabled in secure, internal environments.</Warning> | `true` |
+
+## Setting Up Operations
+
+1. Create a directory to store your GraphQL operations as specified in your `operations_dir` configuration.
+2. Add `.graphql` files containing named GraphQL operations.
```

Comment:
We should think of the implications of using .graphql files.
We need to think about use cases where we want to attach more information to a response, e.g. adding some text to the response.
Should we do this in the form of directives on the GraphQL operation or should we use JSON instead?

Reply:
Good idea. I don't see this as part of this iteration but ideally this can be done on the schema level to encourage schema collaboration. Right now, I do consider root field descriptions from the schema as tool description, as well as type descriptions from types when building the input json schema.

Reply:
Also when someone select multiple root fields. I do join both description together for more context.